go/rep_replication.go

```diff
@@ -0,0 +1,123 @@
+package pb
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ipfs/go-cid"
+)
+
+const header = "rtrade-replication v0"
+
+// New creates a new Replication file, with the correct header
+func New() *Replication {
+	return &Replication{
+		Header: header,
+	}
+}
+
+// Valid checks if this is an Valid Replication file, with:
+// - correct header (if strict, then no unknown extensions),
+// - at least one CID,
+// - if strict, then equal or more servers than required for replication, and
+// - replication factor greater than 1.
+// reports the first error found.
+//
+// If strict, then invalid CIDs, Servers, and unknown protocol extensions are also not allowed.
+func (r *Replication) Valid(strict bool) error {
+	if _, err := r.GetHeaderString(strict); err != nil {
+		return err
+	}
+
+	if strict && haveUnrecognized(r) {
+		return errors.New("found unrecognized data")
+	}
+
+	CIDs, err := r.GetCIDs()
+	if (strict || len(CIDs) == 0) && err != nil {
+		return err
+	}
+	as := ServerSourcesToAddrInfos(r.Servers)
+	if len(as) < int(r.GetReplicationFactor()) {
+		return fmt.Errorf("not enough servers list %v of %v", len(as), r.GetReplicationFactor())
+	}
+	for _, a := range as {
+		if _, err := a.GetID(); err != nil {
+			if strict {
+				return err
+			}
+			continue
+		}
```

Comment:
can be simplified to:

```Go
if _, err := a.GetID(); err != nil && strict {
    return err
} else if err != nil && !strict {
    continue
}
````

Reply:
Rereading this code, I think I can move the strict check outside of this loop. I think I wanted to do a count of well-formed server addresses, but that's unnecessary and not implemented right now.